packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,39 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+export function getVarSize(value: any): number {
+  if (typeof value === "object" && value instanceof HexString) {
+    const size = value.byteLength;
+    return getVarSize(size) + size;
+  } else if (typeof value === "number") {
+    if (value < 0xfd) return 1;
+    else if (value <= 0xffff) return 3;
+    else return 5;

```

Comment:
The last time I implemented VarInt, there was the case of 9 bytes (see StringStream.readVarInt). Probably also requires us to throw if the number is too big?

Reply:
this part of the implementation is identical to NEO's
https://github.com/neo-project/neo/blob/ab7ec04925c1e10c9338e3deb8880e93b75ff306/src/neo/IO/Helper.cs#L99-L107

Reply:
You can read bigger VarInts than 5 bytes.
https://github.com/neo-project/neo/blob/ab7ec04925c1e10c9338e3deb8880e93b75ff306/src/neo/IO/Helper.cs#L211

Reply:
what you linked is for reading byte arrays, not for reading a number